dev/io.openliberty.security.authorization.internal.jacc.3.0/src/io/openliberty/security/authorization/jacc/internal/proxy/ProviderServiceProxyImpl.java

```diff
@@ -85,50 +112,53 @@ public String getFactoryName() {
     }
 
     protected void activate(ComponentContext cc) {
+        locationAdminRef.activate(cc);
         jaccProviderService.activate(cc);
     }
 
     protected void deactivate(ComponentContext cc) {
+        // unset PolicyFactory so we do not leak the user feature classloader for the ProviderService
+        PolicyFactory.setPolicyFactory(null);
+
         jaccProviderService.deactivate(cc);
+        locationAdminRef.deactivate(cc);
+
     }
 
     private void initializeSystemProperties(ServiceReference<ProviderService> reference) {
         Object obj = reference.getProperty(PolicyFactory.FACTORY_NAME);
         if (obj != null && obj instanceof String) {
-            policyName = (String) obj;
+            policyFactoryName = (String) obj;
         }
 
         obj = reference.getProperty(PolicyConfigurationFactory.FACTORY_NAME);
         if (obj != null && obj instanceof String) {
             factoryName = (String) obj;
         }
         if (tc.isDebugEnabled())
-            Tr.debug(tc, "Meta data : policyName : " + policyName + " factoryName : " + factoryName);
+            Tr.debug(tc, "Meta data : policyFactoryName : " + policyFactoryName + " factoryName : " + factoryName);
 
-        originalSystemPolicyName = null;
+        originalSystemPolicyFactoryName = null;
         originalSystemFactoryName = null;
 
-        String systemPolicyName = System.getProperty(PolicyFactory.FACTORY_NAME);
+        String systemPolicyFactoryName = System.getProperty(PolicyFactory.FACTORY_NAME);
 
         String systemFactoryName = System.getProperty(PolicyConfigurationFactory.FACTORY_NAME);
 
         if (tc.isDebugEnabled()) {
-            Tr.debug(tc, "System properties : policyName : " + systemPolicyName + " factoryName : " + systemFactoryName);
+            Tr.debug(tc, "System properties : policyFactoryName : " + systemPolicyFactoryName + " factoryName : " + systemFactoryName);
         }
-        if (systemPolicyName == null) {
-            if (policyName != null) {
-                System.setProperty(PolicyFactory.FACTORY_NAME, policyName);
-            } else if (policyName == null) {
-                Tr.error(tc, "JACC_POLICY_IS_NOT_SET");
-                return;
```

Comment:
Does this scenario no longer apply?

Reply:
Correct.  If it is not set we will just use our internal one and not wrap the one provided.  Looking at this it looks like I didn't mean for us to have the ProviderService provide a PolicyFactory and only honor the System.getProperty one.

Reply:
Actually it is written to be able to honor either one and then check for consistency.  We can decide if we want to have the ProviderService give a PolicyFactory or not.  Some of this gets into the testing that will be done and then will determine how things work.